vector/src/main/java/im/vector/activity/VectorRoomActivity.java

```diff
@@ -721,12 +721,26 @@ public boolean onEditorAction(TextView textView, int actionId, KeyEvent keyEvent
             }
         });
 
+        mRoom = mSession.getDataHandler().getRoom(roomId, false);
+
+        mEditText.initAutoCompletion(mSession, (null != mRoom) ? mRoom.getRoomId() : null);
+        mEditText.initAutoCompletionCommandLine(mSession , null);
         mEditText.setAddColonOnFirstItem(true);
 
         mEditText.addTextChangedListener(new TextWatcher() {
             @Override
             public void afterTextChanged(android.text.Editable s) {
                 if (null != mRoom) {
+                    if (mEditText.getText().toString().startsWith("@")) {
+                        mEditText.setAdapter(mEditText.mAdapterUser);
```

Comment:
You should maybe add an enum for type and a method to change it like :
setMode(AutoCompletion.Command) and setMode(AutoCompletion.User)
The method will be responsible of setting adapter and threshold when the mode is different from the one already set.